logbook-json/src/main/java/org/zalando/logbook/json/JsonPathBodyFilters.java

```diff
@@ -0,0 +1,107 @@
+        }
+
+        public BodyFilter replace(String replacement) {
+            return replace(replacement, false, String.class);
+        }
+
+        public BodyFilter forceReplace(String replacement) {
+            return replace(replacement, true, String.class);
+        }
+
+        public BodyFilter replace(Number replacement) {
+            return replace(replacement, false, Number.class);
+        }
+
+        public BodyFilter forceReplace(Number replacement) {
+            return replace(replacement, true, Number.class);
+        }
+
+        public BodyFilter replace(String matchRegex, String replacementRegex) {
+            Pattern pattern = Pattern.compile(matchRegex);
+            return (JsonBodyFilter) documentContext -> {
+                Object obj = documentContext.read(jsonPath);
+                if (isPrimitive(obj)) {
+                    replaceWithRegex(obj, pattern, replacementRegex)
+                            .ifPresent(s -> documentContext.set(jsonPath, s));
+                } else if (obj instanceof JSONArray) {
+                    ((JSONArray) obj).replaceAll(o -> {
+                        if (isPrimitive(o)) {
+                            Optional<String> replacement = replaceWithRegex(o, pattern, replacementRegex);
+                            if (replacement.isPresent()) {
+                                return replacement.get();
+                            }
+                        }
+                        return o;
+                    });
+                    documentContext.set(jsonPath, obj);
+                }
+                return documentContext.jsonString();
+            };
+        }
+
+        private  <T> BodyFilter replace(T replacement, Boolean force, Class<T> replacementClass) {
+            return (JsonBodyFilter) documentContext -> {
+                Object obj = documentContext.read(jsonPath);
+                if (obj instanceof JSONArray) {
+                    JSONArray array = ((JSONArray) obj);
+                    array.replaceAll(o -> replacementClass.isAssignableFrom(o.getClass()) || force ? replacement : o);
+                    documentContext.set(jsonPath, array);
+                } else if (replacementClass.isAssignableFrom(obj.getClass())) {
+                    documentContext.set(jsonPath, replacement);
+                } else if (force) {
+                    documentContext.set(jsonPath, replacement);
+                }
+                return documentContext.jsonString();
+            };
+        }
+
+        private Optional<String> replaceWithRegex(Object obj, Pattern pattern, String replacementRegex) {
+            String stringRepresentation = obj.toString();
+            Matcher matcher = pattern.matcher(stringRepresentation);
+            if (matcher.find()) {
+                return Optional.of(matcher.replaceAll(replacementRegex));
+            }
+            return Optional.empty();
+        }
+
+        private boolean isPrimitive(Object obj) {
```

Comment:
Could this be simplified by just calling toString/toText (not sure if that exists here) to just operate on primitives as strings?

Reply:
This method doesn't actually convert, it only checks if object is primitive. We need this check to be sure, that specified jsonpath doesn't contain some `{}` object.
As you can see, i call `toString()` on that object after check.